### ISSUE TITLE
QSelect doesnt work with clearable and multiple at the same time

### ISSUE THREAD

**Issue by RufusK** [quasarframework/quasar#8092]
**Describe the bug**
In QSelect clearable sets the v-model to null. At the same time QSelect with multiple prop only works with an empty array. The problem becomes apparent when you use toggles or checkboxes in the select options. They all go into indeterminate state when you click on the clear icon.

**Codepen/jsFiddle/Codesandbox (required)**
https://codepen.io/jimzitrone/pen/yLJmopL

**To Reproduce**
1. Go to https://codepen.io/jimzitrone/pen/yLJmopL
2. Select 1 or more options
3. Click on the clear icon on the right side
4. See all toggles in indeterminate state

**Expected behavior**
Clearable on QSelect with multiple prop should set v-model to empty array instead of null

**Platform (please complete the following information):**
OS: Win 10
Node: 14
NPM: 6.14.8
Yarn:
Browsers: Tested in Chrome (but surely everywhere)
iOS:
Android:
Electron:

![image](https://user-images.githubusercontent.com/11466451/99978140-dd43e480-2da5-11eb-8b50-91d806256e68.png)

**Comment by gaw89** [comment]
@RufusK, I'm not sure if this is a bug, per se, or if it's simply unexpected behavior that should be more clearly noted in the docs.  If you have an immediate need to solve this issue while the quasar team works on it, a simple approach is to use `:value` and `@input` instead of `v-model`.  You can then intercept the `@clear` event to reset your model back to an empty array as in [this codepen](https://codepen.io/gaw89/pen/vYKodJP) and below:

```html
<q-select
  filled
  :value="model"
  @input="v => model = v"
  @clear="model = []"
  :options="options"
  label="Multi with toggle"
  multiple
  emit-value
  map-options
  clearable
>
```

**Comment by pdanpdan** [comment]
Hello.
This is not about QSelect, but about QToggle - when you use it on an array model it will toggle the values in array, but else it will toggle the model.

You cannot use QToggle like this in QSelect.
Use `<q-toggle :value="scope.selected" @input="scope.toggleOption(scope.opt)"></q-toggle>`
Check here: https://codepen.io/pdanpdan/pen/GRqVxbv?editors=1010

**Comment by RufusK** [comment]
Hi, I am pretty sure that this is a bug. Muli-Select always uses an array as model. It would be very easy to check for "multiple" prop and in that case set the model to [] instead of null on clear as a default. There is no documentation that you have to use toggles and checkboxes like this in selects, not even in the code example https://quasar.dev/vue-components/select#Example--Object-options it is done like this. I think props should usually be composable.

**Comment by pdanpdan** [comment]
Ups, the example is bad, it should be fixed. Reopening to track the documentation fix.

**Comment by RufusK** [comment]
Please give it a moment to consider making a fix to make clearable and multiple work together natively instead of just fixing the documentation. I am happy to make a pull request.

**Comment by pdanpdan** [comment]
The example in docs is wrong anyway. It's a totally wrong way to do it.
About changing the way `clearable` work check here: https://github.com/quasarframework/quasar/issues/4182#issuecomment-631149576

**Comment by RufusK** [comment]
I now understand that there is a deeper problem behind the whole clearable thing. Thanks for changing the documentation. Cheers

**Comment by pdanpdan** [comment]
The fixed examples will be available in the docs one Quasar v1.4.6 is released
